rust/logging/src/lib.rs

```diff
@@ -77,18 +77,36 @@ pub fn test_global(directives: &str) {
 /// - warn events are reported as sentry messages
 /// - info events are captured as breadcrumbs (and submitted together with warns & errors)
 ///
-/// # Spans
+/// ## Telemetry events
+///
+/// This layer configuration supports a special `telemetry` event.
+/// Telemetry events are events logged on the `TRACE` level for the `telemetry` target.
+/// They are sampled at a rate of 1%.
+/// The idea here is that some events logged via `tracing` should not necessarily end up in the users log file.
+/// Yet, if they happen a lot, we still want to know about them.
+/// Coupling the `telemetry` target to the `TRACE` level pretty much prevents these events from ever showing up in log files.
+/// By sampling them, we prevent flooding Sentry with lots of these logs.
+///
+/// ## Spans
 ///
 /// The default span-filter captures all spans with level INFO, WARN and ERROR as sentry "transactions".
 pub fn sentry_layer<S>() -> sentry_tracing::SentryLayer<S>
 where
     S: Subscriber + for<'a> LookupSpan<'a>,
 {
     sentry_tracing::layer()
-        .event_filter(|md| match *md.level() {
+        .event_filter(move |md| match *md.level() {
             tracing::Level::ERROR => EventFilter::Exception,
             tracing::Level::WARN => EventFilter::Event,
             tracing::Level::INFO => EventFilter::Breadcrumb,
+            tracing::Level::TRACE if md.target() == "telemetry" => {
+                // rand::random generates floats in the range of [0, 1).
+                if rand::random::<f32>() < 0.01 {
```

Comment:
Should we change this to a unfiorm 1..100 and  do `== 1` to prevent artifacts from floats non-uniformity?

Reply:
Although according to ChatGPT the deviation from 1% is only 0.0063% so you might as well leave it as is

Reply:
`rand::random` already gives me a convenient 0..1 range so I figured I can just use that directly. Happy to change it to 1..100 if you want!